enumerable_methods.rb

```diff
@@ -54,94 +66,147 @@ def my_select
+    if cls.nil?
+      if arr.is_a? Array
+        i = 0
+        while i < arr.length
+          if block_given?
+            return true if yield(arr[i])
+          else
+            return true if arr[i]
+          end
+          i += 1
+        end
+        false
+      else
+        puts 'Error: Parameter is not an array!'
+      end
+    else
       i = 0
+      cls.class == Integer.class ? cls = cls : cls = cls.class
       while i < arr.length
-        return true if yield(arr[i])
+        return true if arr[i].class == cls
 
         i += 1
       end
       false
-    else
-      puts 'Error: Parameter is not an array!'
     end
   end
 
-  def my_none?
-    return to_enum unless block_given?
-
+  def my_none?(cls = nil)
     arr = self
-    if arr.is_a? Array
+    if cls.nil?
+      if arr.is_a? Array
+        x = true
+        i = 0
+        while i < arr.length
+          if block_given?
+            x = false if yield(arr[i])
+          else
+            x = false if arr[i]
+          end
+          i += 1
+        end
+        x
+      else
+        puts 'Error: Parameter is not an array!'
+      end
+    else
       x = true
       i = 0
+      cls.class == Integer.class ? cls = cls : cls = cls.class
+      puts cls
       while i < arr.length
-        x = false if yield(arr[i])
+        x = false if arr[i].class == cls
         i += 1
       end
       x
-    else
-      puts 'Error: Parameter is not an array!'
     end
   end
 
-  def my_count
-    return to_enum unless block_given?
-
+  def my_count(item = nil)
     arr = self
-    if arr.is_a? Array
-      x = 0
-      i = 0
-      while i < arr.length
-        x += 1 if yield(arr[i])
-        i += 1
+    if item.nil?
+      if arr.is_a? Array
+        x = 0
+        i = 0
+        while i < arr.length
+          if block_given?
+            x += 1 if yield(arr[i])
+          else
+            x += 1 if arr[i]
+          end
+          i += 1
+        end
+        x
+      else
+        puts 'Error: Parameter is not an array!'
       end
-      x
     else
-      puts 'Error: Parameter is not an array!'
+      if arr.is_a? Array
+        x = 0
+        i = 0
+        while i < arr.length
+          x += 1 if item == arr[i]
+          i += 1
+        end
+        x
+      else
+        puts 'Error: Parameter is not an array!'
+      end
     end
   end
 
-  def my_inject
-    return to_enum unless block_given?
-
+  def my_inject(initial = nil, sym = nil)
     arr = self
-    if arr.is_a? Array
-      x = arr[0]
-      i = 1
-      while i < arr.length
-        x = yield(x, arr[i])
-        i += 1
-      end
-      x
-    else
-      puts 'Error: Parameter is not an array!'
+    arr = arr.to_a unless arr.class == Array
+    initial.nil? ? (x = arr[0]) : (x = initial + arr[0])
+    i = 1
+    while i < arr.length
+      x = if block_given?
+            yield(x, arr[i])
+          else
+            [x, arr[i]]
+          end
+      i += 1
     end
+    x
   end
```

Comment:
- [ ] combines all elements of enum by applying a binary operation, specified by a block
- [ ] when a symbol is specified combines each element of the collection by applying the symbol as a named method